harness/determined/searcher/searchers.py

```diff
@@ -0,0 +1,543 @@
+import dataclasses
```

Comment:
I don't think we want to ship RandomSearchMethod with harness. It belongs in tests. So I suggest moving those classes, either into harness/tests/test_custom_searcher.py or into a new file harness/tests/search_methods.py.